ex4/ex4.c

```diff
@@ -11,6 +11,29 @@
 int main(void)
 {
     // Your code here    
+    int rc = fork();
+    // ------------------------------------------------ child process starts executing here
+    if (rc < 0) {    // fork failed; exit
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (rc == 0) {    // child process satisfies this branch
+        printf("hello, child here (pid: %d) \n", (int) getpid());
+        char *myargs[2];    // allocate an array of chars to hold 3 bytes
+        // `strdup` duplicates the given input string 
+        myargs[0] = strdup("/bin/ls");      // pass the name of the program we want to run as the first array element
```

Comment:
I'm pretty sure you need to #include string.h to use the strdup function correctly. I am getting a warning about this when I compile. This may be why it ends up seg faulting.